cpp/src/arrow/util/compression_lz4.cc

```diff
@@ -41,13 +41,20 @@ namespace util {
 
 namespace {
 
+constexpr int kLZ4MinCompressionLevel = 1;
+constexpr int kLZ4DefaultCompressionLevel = 1;
+
 static Status LZ4Error(LZ4F_errorCode_t ret, const char* prefix_msg) {
   return Status::IOError(prefix_msg, LZ4F_getErrorName(ret));
 }
 
 static LZ4F_preferences_t DefaultPreferences() {
-  LZ4F_preferences_t prefs;
-  memset(&prefs, 0, sizeof(prefs));
+  LZ4F_preferences_t prefs = LZ4F_INIT_PREFERENCES;
```

Comment:
Please add a comment explaining this macro (or at least I assume it's a macro and not a constant). If it isn't completely trivial, I'd recommend ensuring that it is only evaluated once:
```suggestion
  static LZ4F_preferences_t prefs = LZ4F_INIT_PREFERENCES;
```

Reply:
Actually it seems that using the memset is also fine, so I reverted back to using that